src/foundry/foundry.js/clientDocuments/user.d.ts

```diff
@@ -10,7 +13,135 @@ declare global {
    */
-  class User extends ClientDocumentMixin(foundry.documents.BaseUser) {}
+  class User extends ClientDocumentMixin(foundry.documents.BaseUser) {
+    /**
+     * Track whether the user is currently active in the game
+     */
+    active: boolean;
+
+    /**
+     * Track references to the current set of Tokens which are targeted by the User
+     */
+    targets: UserTargets;
+
+    /**
+     * Track the ID of the Scene that is currently being viewed by the User
+     */
+    viewedScene: string | null;
+
+    /**
+     * Return the User avatar icon or the controlled actor's image
+     */
+    get avatar(): string;
+
+    /**
+     * Return the Actor instance of the user's impersonated character (or undefined)
+     */
+    get character(): ConfiguredDocumentClass<typeof Actor> | undefined;
+
+    /**
+     * A convenience shortcut for the permissions object of the current User
+     */
+    get permissions(): UserData['permissions'];
+
+    /**
+     * A flag for whether the current User is a Trusted Player
+     */
+    get isTrusted(): boolean;
+
+    /**
+     * A flag for whether this User is the connected client
+     */
+    get isSelf(): boolean;
+
+    /**
+     * Assign a Macro to a numbered hotbar slot between 1 and 50
+     * @param macro    - The Macro entity to assign
+     * @param slot     - A specific numbered hotbar slot to fill
+     * @param fromSlot - An optional origin slot from which the Macro is being shifted
+     * @returns A Promise which resolves once the User update is complete
+     */
+    assignHotbarMacro(macro: Macro | null, slot: string | number, { fromSlot }?: { fromSlot: number }): Promise<this>;
+
+    /**
+     * Assign a specific boolean permission to this user.
+     * Modifies the user permissions to grant or restrict access to a feature.
+     *
+     * @param permission - The permission name from USER_PERMISSIONS
+     * @param allowed    - Whether to allow or restrict the permission
+     */
+    assignPermission(permission: keyof typeof CONST.USER_PERMISSIONS, allowed: boolean): Promise<this>;
+
+    /* -------------------------------------------- */
+
+    /**
+     * Submit User activity data to the server for broadcast to other players.
+     * This type of data is transient, persisting only for the duration of the session and not saved to any database.
+     *
+     * @param cursor  - The coordinates of the user's cursor
+     * @param focus   - Is the user pulling focus to the cursor coordinates?
+     * @param ping    - Is the user emitting a ping at the cursor coordinates?
+     * @param ruler   - Serialized Ruler coordinate data in JSON format
+     * @param sceneId - The id of the Scene currently being viewed by the User
+     * @param targets - An id of Token ids which are targeted by the User
+     */
+    broadcastActivity({
+      cursor,
+      focus,
+      ping,
+      ruler,
+      sceneId,
+      targets
+    }: {
+      cursor?: boolean;
```

Comment:
I think it's worth creating a type for that. Also you can move the documentation of individual properties to that type.